src/interfaces/element-props.ts

```diff
@@ -1,9 +1,32 @@
+export const enum Phases {
+  Gas = 'Gas',
+  Liquid = 'Liquid',
+  Solid = 'Solid',
+}
```

Comment:
Use as const assertion for consistency

Reply:
why  I need to use as const assertion if I have already defined the enum 'const' . we have used as const assertion on dictionary where we were assigning const dictionary to const variable , while here I have already defined it const .

Reply:
@qur786 We are typically avoiding the use of TS `const enums` in our projects/repositories.

Instead, the pattern we have adopted is:

```ts
export const Phase = {
  Gas: 'gas',
  Liquid: 'liquid', // notice the casing
  Solid: 'solid',
} as const;
export type Phase = typeof Phase[keyof typeof Phase]; // notice the same names
```

`const enum`s don't play along well with the [`isolatedModules: true`](https://www.typescriptlang.org/tsconfig#isolatedModules) setting in the `tsconfig`. Earlier, it was disabled in this repo, but now I have enabled it since that setting has some benefits associated while using Babel compilers (which our end-users may use instead of using TSC). Our other projects already use this setting.

Infact, TS also suggests using `as const` instead of `const enum` where possible [here](https://www.typescriptlang.org/docs/handbook/enums.html#objects-vs-enums).

Reply:
Thank you @maneetgoyal sir for the explanation.